"Failed to Import TRX File"

Rider 251 Dev, dran&drop a file:
![Image](https://github.com/user-attachments/assets/3318e1fb-e56a-42ef-8e76-4baeda7bcf82)

```
Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.

--- EXCEPTION #1/2 [MissingMethodException]
Message = “Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.”
StackTraceString = “
  at Rider.Plugins.TrxPlugin.RealElementsFinder.FindRealElement(UnitTestResult result)
     at Rider.Plugins.TrxPlugin.TrxManager.<>c__DisplayClass15_0.<DisplayResults>b__1(IUnitTestTransaction tx) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 240
     at JetBrains.ReSharper.UnitTestFramework.Persistence.LevelDb.UnitTestTransaction.CommitCore() in CommitCore.il:line IL_0032 mvid ECDE
     at JetBrains.ReSharper.UnitTestFramework.Persistence.LevelDb.UnitTestTransaction.Commit() in Commit.il:line IL_0043 mvid ECDE
     at System.Threading.Tasks.Task`1.InnerInvoke() in InnerInvoke.il:line IL_000F mvid 243A
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A or RunInternal.il:line IL_0080 mvid 243A
  --- End of stack trace from previous location ---
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A or RunInternal.il:line IL_0080 mvid 243A
     at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in ExecuteWithThreadLocal.il:line IL_0096 mvid 243A
  --- End of stack trace from previous location ---
     at Rider.Plugins.TrxPlugin.TrxManager.DisplayResults(CancellationToken ct, List`1 results, String id, String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 230
     at Rider.Plugins.TrxPlugin.TrxManager.HandleTrx(String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 207
”
ExceptionPath = Root.InnerException
ClassName = System.MissingMethodException
HResult = COR_E_MISSINGMETHOD=80131513
Source = Rider.Plugins.TrxPlugin

--- Outer ---

--- EXCEPTION #2/2 [LoggerException]
Message = “Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.”
InnerException = “Exception #1 at Root.InnerException”
StackTraceString = “
  at JetBrains.Util.ILoggerEx.LogException(ILogger logger, LoggingLevel level, Exception exception, ExceptionOrigin exceptionOrigin, String comment) in LogException.il:line IL_002E mvid 607D
     at JetBrains.Util.ILoggerEx.Error(ILogger logger, Exception exception, ExceptionOrigin origin, String comment) in Error.il:line IL_0000 mvid 607D
     at Rider.Plugins.TrxPlugin.TrxManager.HandleTrx(String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 207
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in ExecutionContextCallback.il:line IL_0000 mvid 243A
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in MoveNext.il:line IL_0040 mvid 243A
     at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask) in RunCallback.il:line IL_0013 mvid 243A
     at System.Threading.Tasks.Task.RunContinuations(Object continuationObject) in RunContinuations.il:line IL_0081 mvid 243A
     at System.Threading.Tasks.Task.FinishSlow(Boolean userDelegateExecute) in FinishSlow.il:line IL_0003 mvid 243A
     at System.Threading.Tasks.Task.TrySetException(Object exceptionObject) in TrySetException.il:line IL_0022 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception exception, Task`1& taskField) in SetException.il:line IL_0025 mvid 243A
     at Rider.Plugins.TrxPlugin.TrxManager.DisplayResults(CancellationToken ct, List`1 results, String id, String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 230
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in MoveNext.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcherSynchronizationContext.<>c__DisplayClass2_0.<Post>b__1() in JetDispatcherSynchronizationContext.il:line IL_0000 mvid 607D
     at JetBrains.Util.Concurrency.ExecutionContextEx.WithinExecutionContext(Object o) in WithinExecutionContext.il:line IL_002F mvid 607D
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_01EF mvid 607D
     at JetBrains.Threading.JetDispatcher.ProcessQueue(Int32 nMinBucket, Int32 nProcessQueueGeneration) in ProcessQueue.il:line IL_01D0 mvid 607D
     at JetBrains.Threading.JetDispatcher.<>c__DisplayClass64_0.<ScheduleQueueProcessingOnNativeMarshaller>g__ProcessQueueWithPriority|0() in JetDispatcher.il:line IL_0000 mvid 607D
     at System.Windows.Threading.DispatcherOperation.InvokeDelegateCore() in InvokeDelegateCore.il:line IL_0000 mvid 09FB
     at System.Windows.Threading.DispatcherOperation.InvokeImpl() in InvokeImpl.il:line IL_0062 mvid 09FB
     at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj) in CallbackWrapper.il:line IL_001A mvid 09FB
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state) in Run.il:line IL_0034 mvid 09FB
     at System.Windows.Threading.DispatcherOperation.Invoke() in Invoke.il:line IL_000F mvid 09FB
     at System.Windows.Threading.Dispatcher.ProcessQueue() in ProcessQueue.il:line IL_00E5 mvid 09FB
     at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in WndProcHook.il:line IL_003A mvid 09FB
     at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in WndProc.il:line IL_0023 mvid 09FB
     at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o) in DispatcherCallbackOperation.il:line IL_0036 mvid 09FB
     at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs) in InternalRealCall.il:line IL_0076 mvid 09FB
     at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler) in TryCatchWhen.il:line IL_0002 mvid 09FB
     at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs) in LegacyInvokeImpl.il:line IL_00BC mvid 09FB
     at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam) in SubclassWndProc.il:line IL_010A mvid 09FB
     at JetBrains.Interop.WinApi.User32Dll.DispatchMessageW(MSG* lpmsg)
     at JetBrains.Interop.WinApi.User32Dll.DispatchMessageW(MSG* lpmsg)
     at JetBrains.Util.Concurrency.WinDispatcherImpl.InternalPumpMessagesOnce() in InternalPumpMessagesOnce.il:line IL_0038 mvid 607D
     at JetBrains.Threading.RunsAndSleeps.RunOrSleep(Func`1 FCondition, IProperty`1 propCondition, TimeSpan timeout, OnWaitTimeout ontimeout, RunOrSleepFlags flags) in RunOrSleep.il:line IL_03B2 mvid 607D
     at JetBrains.Threading.JetDispatcher.Run(IProperty`1 condition, TimeSpan timeout, Boolean bThrowOnTimeout) in Run.il:line IL_0005 mvid 607D
     at JetBrains.Application.Environment.IJetHostEx.RunHostMessageLoop(IComponentContainer containerEnv) in RunHostMessageLoop.il:line IL_000E mvid 4B95
     at JetBrains.Application.Environment.HostParameters.MessagePumpMainLoopHostMixin.JetBrains.Application.Environment.HostParameters.IRunMainLoopHostMixin.RunMainLoop(ComponentContainer containerEnv)
     at JetBrains.Application.Environment.HostParameters.JetHostParametersCaller.RunMainLoop(ComponentContainer containerEnv) in RunMainLoop.il:line IL_009F mvid 4B95
     at JetBrains.Application.Environment.JetEnvironment.<>c__DisplayClass12_0.<RunUntilShutdown>b__0() in JetEnvironment.il:line IL_0000 mvid 4B95
     at JetBrains.Util.Concurrency.ExecutionContextEx.WithinExecutionContext(Object o) in WithinExecutionContext.il:line IL_002F mvid 607D
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_01EF mvid 607D
     at JetBrains.Threading.JetDispatcher.Invoke(OuterLifetime lifetime, String name, Action action, TaskPriority priority, Nullable`1 limit) in Invoke.il:line IL_0000 mvid 607D or Invoke.il:line IL_0073 mvid 607D
     at JetBrains.Threading.JetDispatcher.Invoke(String name, Action action, TaskPriority priority) in Invoke.il:line IL_0000 mvid 607D or Invoke.il:line IL_0073 mvid 607D
     at JetBrains.Application.Environment.JetEnvironment.RunUntilShutdown() in RunUntilShutdown.il:line IL_0044 mvid 4B95
     at JetBrains.Application.Environment.JetEnvironment.CreateAndRun(JetHostParameters hostparams) in CreateAndRun.il:line IL_000B mvid 4B95
     at JetBrains.Application.Environment.JetHostItems.CreateAndRun(Full input) in CreateAndRun.il:line IL_0000 mvid 4B95
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.MainX(Assembly assembly, HostInfo hostInfo) in MainX.il:line IL_00B3 mvid 82A8
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.<>c.<Main>b__4_0(Lifetime lifetime) in RiderBackendProgram.il:line IL_002B mvid 82A8
     at JetBrains.Lifetimes.Lifetime.Using[T](Func`2 action) in Using.il:line IL_0014 mvid 2A6A
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.Main() in Main.il:line IL_005D mvid 82A8
”
ExceptionPath = Root
ClassName = JetBrains.Util.LoggerException
HResult = COR_E_APPLICATION=80131600


com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: com.jetbrains.rd.platform.diagnostics.BackendException: Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.

--- EXCEPTION #1/2 [MissingMethodException]
Message = “Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.”
StackTraceString = “
  at Rider.Plugins.TrxPlugin.RealElementsFinder.FindRealElement(UnitTestResult result)
     at Rider.Plugins.TrxPlugin.TrxManager.<>c__DisplayClass15_0.<DisplayResults>b__1(IUnitTestTransaction tx) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 240
     at JetBrains.ReSharper.UnitTestFramework.Persistence.LevelDb.UnitTestTransaction.CommitCore() in CommitCore.il:line IL_0032 mvid ECDE
     at JetBrains.ReSharper.UnitTestFramework.Persistence.LevelDb.UnitTestTransaction.Commit() in Commit.il:line IL_0043 mvid ECDE
     at System.Threading.Tasks.Task`1.InnerInvoke() in InnerInvoke.il:line IL_000F mvid 243A
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A or RunInternal.il:line IL_0080 mvid 243A
  --- End of stack trace from previous location ---
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A or RunInternal.il:line IL_0080 mvid 243A
     at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in ExecuteWithThreadLocal.il:line IL_0096 mvid 243A
  --- End of stack trace from previous location ---
     at Rider.Plugins.TrxPlugin.TrxManager.DisplayResults(CancellationToken ct, List`1 results, String id, String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 230
     at Rider.Plugins.TrxPlugin.TrxManager.HandleTrx(String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 207
”
ExceptionPath = Root.InnerException
ClassName = System.MissingMethodException
HResult = COR_E_MISSINGMETHOD=80131513
Source = Rider.Plugins.TrxPlugin

--- Outer ---

--- EXCEPTION #2/2 [LoggerException]
Message = “Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.”
InnerException = “Exception #1 at Root.InnerException”
StackTraceString = “
  at JetBrains.Util.ILoggerEx.LogException(ILogger logger, LoggingLevel level, Exception exception, ExceptionOrigin exceptionOrigin, String comment) in LogException.il:line IL_002E mvid 607D
     at JetBrains.Util.ILoggerEx.Error(ILogger logger, Exception exception, ExceptionOrigin origin, String comment) in Error.il:line IL_0000 mvid 607D
     at Rider.Plugins.TrxPlugin.TrxManager.HandleTrx(String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 207
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in ExecutionContextCallback.il:line IL_0000 mvid 243A
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in MoveNext.il:line IL_0040 mvid 243A
     at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask) in RunCallback.il:line IL_0013 mvid 243A
     at System.Threading.Tasks.Task.RunContinuations(Object continuationObject) in RunContinuations.il:line IL_0081 mvid 243A
     at System.Threading.Tasks.Task.FinishSlow(Boolean userDelegateExecute) in FinishSlow.il:line IL_0003 mvid 243A
     at System.Threading.Tasks.Task.TrySetException(Object exceptionObject) in TrySetException.il:line IL_0022 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception exception, Task`1& taskField) in SetException.il:line IL_0025 mvid 243A
     at Rider.Plugins.TrxPlugin.TrxManager.DisplayResults(CancellationToken ct, List`1 results, String id, String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 230
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in MoveNext.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcherSynchronizationContext.<>c__DisplayClass2_0.<Post>b__1() in JetDispatcherSynchronizationContext.il:line IL_0000 mvid 607D
     at JetBrains.Util.Concurrency.ExecutionContextEx.WithinExecutionContext(Object o) in WithinExecutionContext.il:line IL_002F mvid 607D
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_01EF mvid 607D
     at JetBrains.Threading.JetDispatcher.ProcessQueue(Int32 nMinBucket, Int32 nProcessQueueGeneration) in ProcessQueue.il:line IL_01D0 mvid 607D
     at JetBrains.Threading.JetDispatcher.<>c__DisplayClass64_0.<ScheduleQueueProcessingOnNativeMarshaller>g__ProcessQueueWithPriority|0() in JetDispatcher.il:line IL_0000 mvid 607D
     at System.Windows.Threading.DispatcherOperation.InvokeDelegateCore() in InvokeDelegateCore.il:line IL_0000 mvid 09FB
     at System.Windows.Threading.DispatcherOperation.InvokeImpl() in InvokeImpl.il:line IL_0062 mvid 09FB
     at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj) in CallbackWrapper.il:line IL_001A mvid 09FB
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state) in Run.il:line IL_0034 mvid 09FB
     at System.Windows.Threading.DispatcherOperation.Invoke() in Invoke.il:line IL_000F mvid 09FB
     at System.Windows.Threading.Dispatcher.ProcessQueue() in ProcessQueue.il:line IL_00E5 mvid 09FB
     at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in WndProcHook.il:line IL_003A mvid 09FB
     at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in WndProc.il:line IL_0023 mvid 09FB
     at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o) in DispatcherCallbackOperation.il:line IL_0036 mvid 09FB
     at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs) in InternalRealCall.il:line IL_0076 mvid 09FB
     at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler) in TryCatchWhen.il:line IL_0002 mvid 09FB
     at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs) in LegacyInvokeImpl.il:line IL_00BC mvid 09FB
     at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam) in SubclassWndProc.il:line IL_010A mvid 09FB
     at JetBrains.Interop.WinApi.User32Dll.DispatchMessageW(MSG* lpmsg)
     at JetBrains.Interop.WinApi.User32Dll.DispatchMessageW(MSG* lpmsg)
     at JetBrains.Util.Concurrency.WinDispatcherImpl.InternalPumpMessagesOnce() in InternalPumpMessagesOnce.il:line IL_0038 mvid 607D
     at JetBrains.Threading.RunsAndSleeps.RunOrSleep(Func`1 FCondition, IProperty`1 propCondition, TimeSpan timeout, OnWaitTimeout ontimeout, RunOrSleepFlags flags) in RunOrSleep.il:line IL_03B2 mvid 607D
     at JetBrains.Threading.JetDispatcher.Run(IProperty`1 condition, TimeSpan timeout, Boolean bThrowOnTimeout) in Run.il:line IL_0005 mvid 607D
     at JetBrains.Application.Environment.IJetHostEx.RunHostMessageLoop(IComponentContainer containerEnv) in RunHostMessageLoop.il:line IL_000E mvid 4B95
     at JetBrains.Application.Environment.HostParameters.MessagePumpMainLoopHostMixin.JetBrains.Application.Environment.HostParameters.IRunMainLoopHostMixin.RunMainLoop(ComponentContainer containerEnv)
     at JetBrains.Application.Environment.HostParameters.JetHostParametersCaller.RunMainLoop(ComponentContainer containerEnv) in RunMainLoop.il:line IL_009F mvid 4B95
     at JetBrains.Application.Environment.JetEnvironment.<>c__DisplayClass12_0.<RunUntilShutdown>b__0() in JetEnvironment.il:line IL_0000 mvid 4B95
     at JetBrains.Util.Concurrency.ExecutionContextEx.WithinExecutionContext(Object o) in WithinExecutionContext.il:line IL_002F mvid 607D
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_01EF mvid 607D
     at JetBrains.Threading.JetDispatcher.Invoke(OuterLifetime lifetime, String name, Action action, TaskPriority priority, Nullable`1 limit) in Invoke.il:line IL_0000 mvid 607D or Invoke.il:line IL_0073 mvid 607D
     at JetBrains.Threading.JetDispatcher.Invoke(String name, Action action, TaskPriority priority) in Invoke.il:line IL_0000 mvid 607D or Invoke.il:line IL_0073 mvid 607D
     at JetBrains.Application.Environment.JetEnvironment.RunUntilShutdown() in RunUntilShutdown.il:line IL_0044 mvid 4B95
     at JetBrains.Application.Environment.JetEnvironment.CreateAndRun(JetHostParameters hostparams) in CreateAndRun.il:line IL_000B mvid 4B95
     at JetBrains.Application.Environment.JetHostItems.CreateAndRun(Full input) in CreateAndRun.il:line IL_0000 mvid 4B95
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.MainX(Assembly assembly, HostInfo hostInfo) in MainX.il:line IL_00B3 mvid 82A8
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.<>c.<Main>b__4_0(Lifetime lifetime) in RiderBackendProgram.il:line IL_002B mvid 82A8
     at JetBrains.Lifetimes.Lifetime.Using[T](Func`2 action) in Using.il:line IL_0014 mvid 2A6A
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.Main() in Main.il:line IL_005D mvid 82A8
”
ExceptionPath = Root
ClassName = JetBrains.Util.LoggerException
HResult = COR_E_APPLICATION=80131600

	at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:112)
	at com.jetbrains.rdclient.protocol.LoggerModelManager$RootListener.logException(LoggerModelManager.kt:69)
	at com.jetbrains.rdclient.protocol.LoggerModelManager$RootListener.extensionCreated$lambda$0(LoggerModelManager.kt:36)
	at com.jetbrains.rd.util.reactive.Signal.fire(Signal.kt:32)
	at com.jetbrains.rd.framework.impl.RdSignal$onWireReceived$2.invoke(RdSignal.kt:33)
	at com.jetbrains.rd.framework.impl.RdSignal$onWireReceived$2.invoke(RdSignal.kt:31)
	at com.jetbrains.rd.framework.MessageBroker$RdWireableDispatchHelper$doDispatch$2.invoke(MessageBroker.kt:126)
	at com.jetbrains.rd.framework.MessageBroker$RdWireableDispatchHelper$doDispatch$2.invoke(MessageBroker.kt:119)
	at com.intellij.concurrency.ThreadContext.captureThreadContext$lambda$10(threadContext.kt:425)
	at com.intellij.util.concurrency.ContextCallable.lambda$call$0(ContextCallable.java:74)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:108)
	at com.intellij.util.concurrency.ContextCallable.call(ContextCallable.java:72)
	at com.intellij.concurrency.ThreadContext$captureThreadContext$3.invoke(threadContext.kt:426)
	at com.jetbrains.rdclient.protocol.RdDispatcher.flushAllInternal(RdDispatcher.kt:122)
	at com.jetbrains.rdclient.protocol.RdDispatcher.flushAll(RdDispatcher.kt:99)
	at com.jetbrains.rdclient.protocol.RdDispatcher.flushAllRunnable$lambda$0(RdDispatcher.kt:34)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:240)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:25)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:202)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runIntendedWriteActionOnCurrentThread$lambda$6(AnyThreadWriteThreadingSupport.kt:337)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction$lambda$5(AnyThreadWriteThreadingSupport.kt:256)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWithTemporaryThreadLocal(AnyThreadWriteThreadingSupport.kt:203)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:256)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runIntendedWriteActionOnCurrentThread(AnyThreadWriteThreadingSupport.kt:336)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:854)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:432)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:102)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:102)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:108)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:102)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:117)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:584)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:481)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10$lambda$9(IdeEventQueue.kt:308)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:858)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10(IdeEventQueue.kt:307)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2$lambda$1(IdeEventQueue.kt:952)
	at com.intellij.openapi.application.WriteIntentReadAction.lambda$run$0(WriteIntentReadAction.java:24)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction$lambda$5(AnyThreadWriteThreadingSupport.kt:256)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWithTemporaryThreadLocal(AnyThreadWriteThreadingSupport.kt:203)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:256)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:934)
	at com.intellij.openapi.application.WriteIntentReadAction.compute(WriteIntentReadAction.java:55)
	at com.intellij.openapi.application.WriteIntentReadAction.run(WriteIntentReadAction.java:23)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2(IdeEventQueue.kt:949)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:963)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:109)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:963)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12(IdeEventQueue.kt:302)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:342)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: com.jetbrains.rd.platform.diagnostics.BackendException: Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.

--- EXCEPTION #1/2 [MissingMethodException]
Message = “Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.”
StackTraceString = “
  at Rider.Plugins.TrxPlugin.RealElementsFinder.FindRealElement(UnitTestResult result)
     at Rider.Plugins.TrxPlugin.TrxManager.<>c__DisplayClass15_0.<DisplayResults>b__1(IUnitTestTransaction tx) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 240
     at JetBrains.ReSharper.UnitTestFramework.Persistence.LevelDb.UnitTestTransaction.CommitCore() in CommitCore.il:line IL_0032 mvid ECDE
     at JetBrains.ReSharper.UnitTestFramework.Persistence.LevelDb.UnitTestTransaction.Commit() in Commit.il:line IL_0043 mvid ECDE
     at System.Threading.Tasks.Task`1.InnerInvoke() in InnerInvoke.il:line IL_000F mvid 243A
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A or RunInternal.il:line IL_0080 mvid 243A
  --- End of stack trace from previous location ---
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A or RunInternal.il:line IL_0080 mvid 243A
     at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in ExecuteWithThreadLocal.il:line IL_0096 mvid 243A
  --- End of stack trace from previous location ---
     at Rider.Plugins.TrxPlugin.TrxManager.DisplayResults(CancellationToken ct, List`1 results, String id, String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 230
     at Rider.Plugins.TrxPlugin.TrxManager.HandleTrx(String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 207
”
ExceptionPath = Root.InnerException
ClassName = System.MissingMethodException
HResult = COR_E_MISSINGMETHOD=80131513
Source = Rider.Plugins.TrxPlugin

--- Outer ---

--- EXCEPTION #2/2 [LoggerException]
Message = “Method not found: 'UndoUsingReadLock JetBrains.Application.Threading.IShellLocksEx.UsingReadLock(JetBrains.Application.Threading.IShellLocks)'.”
InnerException = “Exception #1 at Root.InnerException”
StackTraceString = “
  at JetBrains.Util.ILoggerEx.LogException(ILogger logger, LoggingLevel level, Exception exception, ExceptionOrigin exceptionOrigin, String comment) in LogException.il:line IL_002E mvid 607D
     at JetBrains.Util.ILoggerEx.Error(ILogger logger, Exception exception, ExceptionOrigin origin, String comment) in Error.il:line IL_0000 mvid 607D
     at Rider.Plugins.TrxPlugin.TrxManager.HandleTrx(String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 207
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in ExecutionContextCallback.il:line IL_0000 mvid 243A
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in MoveNext.il:line IL_0040 mvid 243A
     at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask) in RunCallback.il:line IL_0013 mvid 243A
     at System.Threading.Tasks.Task.RunContinuations(Object continuationObject) in RunContinuations.il:line IL_0081 mvid 243A
     at System.Threading.Tasks.Task.FinishSlow(Boolean userDelegateExecute) in FinishSlow.il:line IL_0003 mvid 243A
     at System.Threading.Tasks.Task.TrySetException(Object exceptionObject) in TrySetException.il:line IL_0022 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception exception, Task`1& taskField) in SetException.il:line IL_0025 mvid 243A
     at Rider.Plugins.TrxPlugin.TrxManager.DisplayResults(CancellationToken ct, List`1 results, String id, String trxFilePath) in /mnt/agent/work/937327211b0ff7c3/src/dotnet/Rider.Plugins.TrxPlugin/TrxManager.cs:line 230
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in MoveNext.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcherSynchronizationContext.<>c__DisplayClass2_0.<Post>b__1() in JetDispatcherSynchronizationContext.il:line IL_0000 mvid 607D
     at JetBrains.Util.Concurrency.ExecutionContextEx.WithinExecutionContext(Object o) in WithinExecutionContext.il:line IL_002F mvid 607D
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_01EF mvid 607D
     at JetBrains.Threading.JetDispatcher.ProcessQueue(Int32 nMinBucket, Int32 nProcessQueueGeneration) in ProcessQueue.il:line IL_01D0 mvid 607D
     at JetBrains.Threading.JetDispatcher.<>c__DisplayClass64_0.<ScheduleQueueProcessingOnNativeMarshaller>g__ProcessQueueWithPriority|0() in JetDispatcher.il:line IL_0000 mvid 607D
     at System.Windows.Threading.DispatcherOperation.InvokeDelegateCore() in InvokeDelegateCore.il:line IL_0000 mvid 09FB
     at System.Windows.Threading.DispatcherOperation.InvokeImpl() in InvokeImpl.il:line IL_0062 mvid 09FB
     at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj) in CallbackWrapper.il:line IL_001A mvid 09FB
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state) in Run.il:line IL_0034 mvid 09FB
     at System.Windows.Threading.DispatcherOperation.Invoke() in Invoke.il:line IL_000F mvid 09FB
     at System.Windows.Threading.Dispatcher.ProcessQueue() in ProcessQueue.il:line IL_00E5 mvid 09FB
     at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in WndProcHook.il:line IL_003A mvid 09FB
     at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in WndProc.il:line IL_0023 mvid 09FB
     at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o) in DispatcherCallbackOperation.il:line IL_0036 mvid 09FB
     at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs) in InternalRealCall.il:line IL_0076 mvid 09FB
     at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler) in TryCatchWhen.il:line IL_0002 mvid 09FB
     at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs) in LegacyInvokeImpl.il:line IL_00BC mvid 09FB
     at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam) in SubclassWndProc.il:line IL_010A mvid 09FB
     at JetBrains.Interop.WinApi.User32Dll.DispatchMessageW(MSG* lpmsg)
     at JetBrains.Interop.WinApi.User32Dll.DispatchMessageW(MSG* lpmsg)
     at JetBrains.Util.Concurrency.WinDispatcherImpl.InternalPumpMessagesOnce() in InternalPumpMessagesOnce.il:line IL_0038 mvid 607D
     at JetBrains.Threading.RunsAndSleeps.RunOrSleep(Func`1 FCondition, IProperty`1 propCondition, TimeSpan timeout, OnWaitTimeout ontimeout, RunOrSleepFlags flags) in RunOrSleep.il:line IL_03B2 mvid 607D
     at JetBrains.Threading.JetDispatcher.Run(IProperty`1 condition, TimeSpan timeout, Boolean bThrowOnTimeout) in Run.il:line IL_0005 mvid 607D
     at JetBrains.Application.Environment.IJetHostEx.RunHostMessageLoop(IComponentContainer containerEnv) in RunHostMessageLoop.il:line IL_000E mvid 4B95
     at JetBrains.Application.Environment.HostParameters.MessagePumpMainLoopHostMixin.JetBrains.Application.Environment.HostParameters.IRunMainLoopHostMixin.RunMainLoop(ComponentContainer containerEnv)
     at JetBrains.Application.Environment.HostParameters.JetHostParametersCaller.RunMainLoop(ComponentContainer containerEnv) in RunMainLoop.il:line IL_009F mvid 4B95
     at JetBrains.Application.Environment.JetEnvironment.<>c__DisplayClass12_0.<RunUntilShutdown>b__0() in JetEnvironment.il:line IL_0000 mvid 4B95
     at JetBrains.Util.Concurrency.ExecutionContextEx.WithinExecutionContext(Object o) in WithinExecutionContext.il:line IL_002F mvid 607D
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in RunInternal.il:line IL_0040 mvid 243A
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_01EF mvid 607D
     at JetBrains.Threading.JetDispatcher.Invoke(OuterLifetime lifetime, String name, Action action, TaskPriority priority, Nullable`1 limit) in Invoke.il:line IL_0000 mvid 607D or Invoke.il:line IL_0073 mvid 607D
     at JetBrains.Threading.JetDispatcher.Invoke(String name, Action action, TaskPriority priority) in Invoke.il:line IL_0000 mvid 607D or Invoke.il:line IL_0073 mvid 607D
     at JetBrains.Application.Environment.JetEnvironment.RunUntilShutdown() in RunUntilShutdown.il:line IL_0044 mvid 4B95
     at JetBrains.Application.Environment.JetEnvironment.CreateAndRun(JetHostParameters hostparams) in CreateAndRun.il:line IL_000B mvid 4B95
     at JetBrains.Application.Environment.JetHostItems.CreateAndRun(Full input) in CreateAndRun.il:line IL_0000 mvid 4B95
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.MainX(Assembly assembly, HostInfo hostInfo) in MainX.il:line IL_00B3 mvid 82A8
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.<>c.<Main>b__4_0(Lifetime lifetime) in RiderBackendProgram.il:line IL_002B mvid 82A8
     at JetBrains.Lifetimes.Lifetime.Using[T](Func`2 action) in Using.il:line IL_0014 mvid 2A6A
     at JetBrains.Rider.Backend.Product.RiderBackendProgram.Main() in Main.il:line IL_005D mvid 82A8
”
ExceptionPath = Root
ClassName = JetBrains.Util.LoggerException
HResult = COR_E_APPLICATION=80131600

	at com.jetbrains.rdclient.protocol.LoggerModelManager$RootListener.fromLogEntry(LoggerModelManager.kt:60)
	at com.jetbrains.rdclient.protocol.LoggerModelManager$RootListener.extensionCreated$lambda$0(LoggerModelManager.kt:29)
	... 62 more

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 ### Changed
+- **(Requirement update!)** Support Rider versions in range \[2025.1 EAP 3; 2025.1.*\].
 - Update the plugin name to clarify the supported technology.
 
 ## [1.0.2] - 2024-11-17

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -188,7 +188,7 @@ tasks {
     withType<Test> {
         classpath -= classpath.filter {
             (it.name.startsWith("localization-") && it.name.endsWith(".jar")) // TODO: https://youtrack.jetbrains.com/issue/IJPL-178084/External-plugin-tests-break-due-to-localization-issues
-                || it.name == "platform-ijent-impl.jar" // TODO: Check after 251 migration
+                || it.name == "cwm-plugin.jar" // TODO: Check after 251 EAP5 release
         }
 
         useTestNG()
@@ -198,6 +198,16 @@ tasks {
         }
         environment["LOCAL_ENV_RUN"] = "true"
     }
+
+    val testRiderPreview by intellijPlatformTesting.testIde.registering {
+        version = libs.versions.riderSdkPreview
+        useInstaller = false
+        task {
+            enabled = libs.versions.riderSdk.get() != libs.versions.riderSdkPreview.get()
+        }
+    }
+
+    check { dependsOn(testRiderPreview.name) }
 
     runIde {
         jvmArgs("-Xmx1500m")

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,10 +1,12 @@
 import com.jetbrains.plugin.structure.base.utils.isFile
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
 import org.jetbrains.changelog.exceptions.MissingVersionException
 import org.jetbrains.intellij.platform.gradle.Constants
 import org.jetbrains.intellij.platform.gradle.tasks.PrepareSandboxTask
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 import kotlin.io.path.absolute
 import kotlin.io.path.isDirectory
+import kotlin.io.path.pathString
 
 plugins {
     alias(libs.plugins.changelog)
@@ -63,6 +65,11 @@ dependencies {
         }
 
         jetbrainsRuntime()
+
+        // TODO: Fix after migration to intellij-platform-gradle-plugin 2.2.2
+        bundledLibrary(provider {
+            project.intellijPlatform.platformPath.resolve("lib/testFramework.jar").pathString
+        })
     }
 }
 
@@ -176,6 +183,20 @@ tasks {
                 if (!file.exists()) throw RuntimeException("File \"$file\" does not exist.")
             }
         }
+    }
+
+    withType<Test> {
+        classpath -= classpath.filter {
+            (it.name.startsWith("localization-") && it.name.endsWith(".jar"))
+                || it.name == "platform-ijent-impl.jar"
+        }
+
+        useTestNG()
+        testLogging {
+            showStandardStreams = true
+            exceptionFormat = TestExceptionFormat.FULL
+        }
+        environment["LOCAL_ENV_RUN"] = "true"
     }
 
     runIde {

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -187,8 +187,8 @@ tasks {
 
     withType<Test> {
         classpath -= classpath.filter {
-            (it.name.startsWith("localization-") && it.name.endsWith(".jar"))
-                || it.name == "platform-ijent-impl.jar"
+            (it.name.startsWith("localization-") && it.name.endsWith(".jar")) // TODO: https://youtrack.jetbrains.com/issue/IJPL-178084/External-plugin-tests-break-due-to-localization-issues
+                || it.name == "platform-ijent-impl.jar" // TODO: Check after 251 migration
         }
 
         useTestNG()

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -1,7 +1,8 @@
 [versions]
-kotlin = "2.0.21"
-rdGen = "2024.3.1"
-riderSdk = "2024.3"
+kotlin = "2.1.10"
+rdGen = "2025.1.1"
+riderSdk = "2025.1-EAP3-SNAPSHOT"
+riderSdkPreview = "2025.1-EAP3-SNAPSHOT"
 
 [libraries]
 kotlinStdLib = { group = "org.jetbrains.kotlin", name = "kotlin-stdlib", version.ref = "kotlin" }

--- a/intellij-updater.json
+++ b/intellij-updater.json
@@ -17,5 +17,6 @@
         "field": "kotlin",
         "kind": "kotlin",
         "versionFlavor": "release"
-    }]
+    }],
+    "prBodyPrefix": "## Maintainer Note\n> [!WARNING]\n> This PR will not trigger CI by default. Please **close it and reopen manually** to trigger the CI.\n>\n> Unfortunately, this is a consequence of the current GitHub Action security model (by default, PRs created automatically aren't allowed to trigger other automation)."
 }

--- a/intellij-updater.json
+++ b/intellij-updater.json
@@ -3,9 +3,12 @@
         "file": "gradle/libs.versions.toml",
         "field": "riderSdk",
         "kind": "rider",
-        "versionFlavor": "release",
-        "versionConstraint": "latestWave",
-        "order": "oldest"
+        "versionFlavor": "eap"
+    }, {
+        "file": "gradle/libs.versions.toml",
+        "field": "riderSdkPreview",
+        "kind": "rider",
+        "versionFlavor": "eap"
     }, {
         "file": "gradle.properties",
         "field": "untilBuildVersion",

--- a/src/test/kotlin/com/jetbrains/rider/plugins/trxplugin/test/cases/BasicTrxImportTest.kt
+++ b/src/test/kotlin/com/jetbrains/rider/plugins/trxplugin/test/cases/BasicTrxImportTest.kt
@@ -1,0 +1,46 @@
+package com.jetbrains.rider.plugins.trxplugin.test.cases
+
+import com.jetbrains.rider.plugins.trxplugin.TrxImportService
+import com.jetbrains.rider.protocol.protocol
+import com.jetbrains.rider.test.asserts.shouldBe
+import com.jetbrains.rider.test.asserts.shouldBeTrue
+import com.jetbrains.rider.test.asserts.shouldNotBeNull
+import com.jetbrains.rider.test.base.BaseTestWithSolution
+import com.jetbrains.rider.test.scriptingApi.runBlockingWithProtocolPumping
+import com.jetbrains.rider.test.scriptingApi.withUtFacade
+import org.testng.annotations.Test
+import java.nio.file.Path
+import kotlin.io.path.Path
+import kotlin.io.path.absolute
+import kotlin.io.path.isRegularFile
+
+class BasicTrxImportTest : BaseTestWithSolution() {
+    override val testSolution = "EmptySolution"
+
+    @Test
+    fun test1TrxImport() {
+        val file = repositoryRoot.value.resolve("src/dotnet/Tests/TestData/test1.trx")
+        file.isRegularFile().shouldBeTrue("File \"$file\" should exist.")
+
+        val service = TrxImportService.getInstance(project)
+        val response = runBlockingWithProtocolPumping(project.protocol, "BasicTrxImportTest") {
+            service.importTrx(file.toString())
+        }
+        response.result.shouldBe("Success")
+        withUtFacade(project) { ut ->
+            val descriptor = ut.waitForAnySession()
+            descriptor.title.shouldBe("test1.trx")
+
+            ut.compareSessionTreeWithGold(descriptor, testGoldFile)
+        }
+    }
+}
+
+private val repositoryRoot = lazy {
+    val cwd = Path(".").absolute()
+    var current: Path? = cwd
+    while (current != null && !current.resolve("settings.gradle.kts").isRegularFile()) {
+        current = current.parent
+    }
+    return@lazy current.shouldNotBeNull("repository root not found on top of \"$cwd\".")
+}

--- a/testData/com/jetbrains/rider/plugins/trxplugin/test/cases/BasicTrxImportTest/test1TrxImport/gold/test1TrxImport.gold
+++ b/testData/com/jetbrains/rider/plugins/trxplugin/test/cases/BasicTrxImportTest/test1TrxImport/gold/test1TrxImport.gold
@@ -1,0 +1,5 @@
+-ROOT
+ -Namespace
+  -ParentTest
+   ChildTest1
+   ChildTest2

--- a/testData/solutions/EmptySolution/EmptySolution.sln
+++ b/testData/solutions/EmptySolution/EmptySolution.sln
@@ -3,8 +3,6 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 17
 VisualStudioVersion = 17.0.31903.59
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MyTestProject", "MyTestProject\MyTestProject.csproj", "{4E1EC6B2-3BEB-4D00-B152-72380DF814E2}"
-EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
@@ -14,9 +12,5 @@ Global
 		HideSolutionNode = FALSE
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{4E1EC6B2-3BEB-4D00-B152-72380DF814E2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{4E1EC6B2-3BEB-4D00-B152-72380DF814E2}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{4E1EC6B2-3BEB-4D00-B152-72380DF814E2}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{4E1EC6B2-3BEB-4D00-B152-72380DF814E2}.Release|Any CPU.Build.0 = Release|Any CPU
 	EndGlobalSection
 EndGlobal

--- a/testData/solutions/MyTestSolution/MyTestProject/Class1.cs
+++ b/testData/solutions/MyTestSolution/MyTestProject/Class1.cs
@@ -1,6 +1,0 @@
-﻿namespace MyTestProject;
-
-public class Class1
-{
-
-}

--- a/testData/solutions/MyTestSolution/MyTestProject/MyTestProject.csproj
+++ b/testData/solutions/MyTestSolution/MyTestProject/MyTestProject.csproj
@@ -1,9 +1,0 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
-  <PropertyGroup>
-    <TargetFramework>net8.0</TargetFramework>
-    <ImplicitUsings>enable</ImplicitUsings>
-    <Nullable>enable</Nullable>
-  </PropertyGroup>
-
-</Project>

